zio-http/src/main/scala/zio/http/endpoint/openapi/JsonSchema.scala

```diff
@@ -0,0 +1,910 @@
+              JsonSchemas(
+                JsonSchema.Object(
+                  Map.empty,
+                  Right(nested.root),
+                  Chunk.empty,
+                ),
+                ref,
+                nested.children,
+              )
+            } else {
+              JsonSchemas(
+                JsonSchema.Object(
+                  Map.empty,
+                  Right(nested.root),
+                  Chunk.empty,
+                ),
+                ref,
+                nested.children + (nested.rootRef.get -> nested.root),
+              )
+            }
+          case Schema.Set(elementSchema, _)               =>
+            arraySchemaMulti(refType, ref, elementSchema)
+        }
+      case Schema.Transform(schema, _, _, _, _)                                              =>
+        fromZSchemaMulti(schema, refType)
+      case Schema.Primitive(_, _)                                                            =>
+        JsonSchemas(fromZSchema(schema, SchemaStyle.Inline), ref, Map.empty)
+      case Schema.Optional(schema, _)                                                        =>
+        fromZSchemaMulti(schema, refType)
+      case Schema.Fail(_, _)                                                                 =>
+        throw new IllegalArgumentException("Fail schema is not supported.")
+      case Schema.Tuple2(left, right, _)                                                     =>
+        val leftSchema  = fromZSchemaMulti(left, refType)
+        val rightSchema = fromZSchemaMulti(right, refType)
+        JsonSchemas(
+          AllOfSchema(Chunk(leftSchema.root, rightSchema.root)),
+          ref,
+          leftSchema.children ++ rightSchema.children,
+        )
+      case Schema.Either(left, right, _)                                                     =>
+        val leftSchema  = fromZSchemaMulti(left, refType)
+        val rightSchema = fromZSchemaMulti(right, refType)
+        JsonSchemas(
+          OneOfSchema(Chunk(leftSchema.root, rightSchema.root)),
+          ref,
+          leftSchema.children ++ rightSchema.children,
+        )
+      case Schema.Lazy(schema0)                                                              =>
+        fromZSchemaMulti(schema0(), refType)
+      case Schema.Dynamic(_)                                                                 =>
+        throw new IllegalArgumentException("Dynamic schema is not supported.")
+    }
+  }
+
+  private def arraySchemaMulti(
+    refType: SchemaStyle,
+    ref: Option[String],
+    elementSchema: Schema[_],
+  ): JsonSchemas = {
+    val nested = fromZSchemaMulti(elementSchema, refType)
+    if (elementSchema.isInstanceOf[Schema.Primitive[_]]) {
+      JsonSchemas(
+        JsonSchema.ArrayType(Some(nested.root)),
+        ref,
+        nested.children,
+      )
+    } else {
+      JsonSchemas(
+        JsonSchema.ArrayType(Some(nested.root)),
+        ref,
+        nested.children + (nested.rootRef.get -> nested.root),
```

Comment:
This crashes with a `None.get` for me.

```suggestion
        nested.children ++ nested.rootRef.map(_ -> nested.root),
```

Reply:
Can you give me a reproducer?